testing.py

```diff
@@ -1110,6 +1113,17 @@ def queue_response(self, response_code, media_type="text/html",
                 headers[k.lower()] = v
         self.responses.append((response_code, headers, content))
 
+    def queue_requests_response(
+            self, response_code, media_type="text/html",
+            other_headers=None, content=''
+    ):
+        """Queue a response of the type produced by HTTP.get_with_timeout."""
+        headers = dict(other_headers or {})
+        if media_type:
+            headers['Content-Type'] = media_type
+        response = MockRequestsResponse(response_code, headers, content)
+        self.responses.append(response)
+
```

Comment:
It's a little annoying that we have two different ways of making HTTP requests (Representation.simple_http_get and HTTP.get_with_timeout), but there's no reason DummyHTTPClient can't work for mocking either technique -- the only difference in most cases is the output that needs to be queued.